### ISSUE TITLE
Potentially revisit Selection in Delegated Identity API

### ISSUE THREAD

**Issue by EItanya** [spiffe/spire#4408]
For a TLDR please see the following slack thread: https://spiffe.slack.com/archives/C7XDP01HB/p1691022146761589

Currently in the DelegatedIdentityAPI, the selection logic works using the same AND logic of all SPIRE selectors, rather than the OR logic of Kubernetes style selectors. AND style selectors make sense when it comes to matching Workloads to Entries, but may make a little less sense here. In the case of the DelegatedIdentity API, the authorized delegate already has access to all of the SVIDS/IDs of the Agent, however, it needs to know ALL of the selectors of all those SVIDs to access them.

The following is an example. If I have the following registration entries: 
```
Entry ID         : f6c0e8d8-90bb-45ca-af5f-6990b3049875
SPIFFE ID        : spiffe://example.org/ns/test/sa/httpbin
Parent ID        : spiffe://example.org/ns/spire/sa/spire-agent
Revision         : 0
X509-SVID TTL    : default
JWT-SVID TTL     : default
Selector         : k8s:ns:test
Selector         : k8s:sa:httpbin

Entry ID         : b2f684fc-cb6f-496d-ba9d-4bb2543a3b16
SPIFFE ID        : spiffe://example.org/ns/test/sa/sleep
Parent ID        : spiffe://example.org/ns/spire/sa/spire-agent
Revision         : 0
X509-SVID TTL    : default
JWT-SVID TTL     : default
Selector         : k8s:ns:test
Selector         : k8s:sa:sleep
```

To get the SVIDs for both of these entries I would need to pass:
`key: 'k8s', val: 'ns:test' && key: 'k8s', val: 'sa:sleep' && key: 'k8s', val: 'sa:httpbin'`

`key: 'k8s', val: 'ns:test'` yields no results.

Given that the authorized delegate has access to all of these SVIDs, it seems that knowing the selectors ahead of time, or connecting to the server to get entries doesn't make a lot of sense.

**Comment by amartinezfayo** [comment]
Thank you @EItanya for filing this issue.
I think that the reasoning behind the need of having the same AND logic as workload attestation is that presumably the authorized delegate knows the selectors that match and describes the workload, so it can provide and SVID.
If the OR logic would be applied here, then clients of the API would subscribe to a set of SVIDs that may or may not belong to the workload with the full set of selectors. How would the authorized delegate know which SVID should deliver to which workload?

**Comment by amartinezfayo** [comment]
Hi @EItanya, just pinging here to know if you had a chance to look at my comment?

**Comment by EItanya** [comment]
Hey @amartinezfayo, sorry. I wanted to take some time to think through a response but got pulled off onto other things. I think for now we are working around the current restriction, but will potentially revisit later if we think it really won't work.

**Comment by amartinezfayo** [comment]
Thank you @EItanya, I'm closing this issue for now. Feel free to reopen it in the future if you need.
